vscode-extension/src/tools/createPromptTemplates.ts

```diff
@@ -0,0 +1,240 @@
+        errors.push(`Failed to create ${template.filename}: ${message}`);
+      }
+    }
+
+    return {
+      success: errors.length === 0,
+      files_created: filesCreated,
+      errors
+    };
+  } catch (error) {
+    const message = error instanceof Error ? error.message : String(error);
+    return {
+      success: false,
+      files_created: filesCreated,
+      errors: [`Failed to create prompt templates: ${message}`]
+    };
+  }
+}
+
+/**
+ * Register the paw_create_prompt_templates language model tool with VS Code.
+ * 
+ * This tool is made available to GitHub Copilot agents via the Language Model API.
+ * When an agent calls this tool, it receives parameters (feature_slug and workspace_path)
+ * and returns a result indicating success or failure along with details of created files.
+ * 
+ * The tool is declared in package.json under contributes.languageModelTools and
+ * registered here at extension activation time.
+ * 
+ * Tool Design Pattern:
+ * - Extension provides procedural tools for reliable operations (file creation)
+ * - Agent provides decision-making logic (slug validation, error handling)
+ * - This separation keeps extension code minimal while leveraging agent intelligence
+ * 
+ * @param context - Extension context for registering disposable resources
+ */
+export function registerPromptTemplatesTool(
+  context: vscode.ExtensionContext
+): void {
+  const tool = vscode.lm.registerTool<CreatePromptTemplatesParams>(
+    'paw_create_prompt_templates',
+    {
+      async prepareInvocation(options, _token) {
+        const { feature_slug } = options.input;
+        return {
+          invocationMessage: `Creating 9 PAW prompt template files for feature: ${feature_slug}`,
+          confirmationMessages: {
+            title: 'Create PAW Prompt Templates',
+            message: new vscode.MarkdownString(
+              `This will create 9 prompt template files in:\n\n` +
+              `\`.paw/work/${feature_slug}/prompts/\`\n\n` +
+              `The following files will be created:\n` +
+              `- 01A-spec.prompt.md\n` +
+              `- 01B-spec-research.prompt.md\n` +
+              `- 02A-code-research.prompt.md\n` +
+              `- 02B-impl-plan.prompt.md\n` +
+              `- 03A-implement.prompt.md\n` +
+              `- 03B-review.prompt.md\n` +
+              `- 04-docs.prompt.md\n` +
+              `- 05-pr.prompt.md\n` +
+              `- 0X-status.prompt.md`
+            )
+          }
+        };
+      },
+      async invoke(options) {
+        const result = await createPromptTemplates(options.input);
+
+        if (result.success) {
+          const details = result.files_created
+            .map(file => `- ${file}`)
+            .join('\n');
+          return new vscode.LanguageModelToolResult([
+            new vscode.LanguageModelTextPart(
+              `Successfully created ${result.files_created.length} prompt template files:\n${details}`
```

Comment:
The tool call result should contain the paths to the template files created, so that the agent knows what they are.